lib/licensed/sources/mix.rb

```diff
@@ -0,0 +1,396 @@
+# frozen_string_literal: true
+require "strscan"
+
+module Licensed
+  module Sources
+    class Mix < Source
+
+      LOCKFILE = "mix.lock"
+
+      # Returns whether a mix.lock is present
+      def enabled?
+        File.exists?(LOCKFILE)
+      end
+
+      def enumerate_dependencies
+        parsed_dependencies
+      end
+
+      private
+
+      # Returns a memoized Array of Dependency instances.
+      def parsed_dependencies
+        @parsed_dependencies ||= find_packages.map do |name, lock_info|
+          convert_package_to_dependency(name, lock_info)
+        end
+      end
+
+      # Returns the parsed mix.lock information as a Hash.
+      def find_packages
+        lockfile = File.read(LOCKFILE)
+        LockfileParser.run(lockfile)
+      end
+
+      # Converts a raw package representation to a dependency.
+      #
+      # name      - The name of the package as a String.
+      # lock_info - The parsed lockfile data for the package as an Array.
+      #
+      # Returns a Dependency, or raises an ArgumentError if unsuccessful.
+      def convert_package_to_dependency(name, lock_info)
+        package_type = lock_info.first
+        case package_type
+        when "git"
+          git_dependency(name, lock_info)
+        when "hex"
+          hex_dependency(name, lock_info)
+        else
+          raise ArgumentError, "Unknown package type in mix.lock: #{package_type}"
+        end
+      end
+
+      # Generate a Dependency for a Git-based package type.
+      #
+      # name      - The name of the package as a String.
+      # lock_info - The parsed lockfile data for the package as an Array.
+      #
+      # Returns a Dependency (possibly with error information).
+      def git_dependency(name, lock_info)
+        # Example: {:git, "https://example.com/path/to/repo.git", "A-SHA-HERE", []},
+        path, errors = check_dep_path(name)
+        if lock_info.length == 4
+          Dependency.new(
+            name: name,
+            version: lock_info[2],
+            path: path,
+            metadata: {
+              "type" => "git",
```

Comment:
The `type` metadata should be an identifier for the package manager that this source enumerates.  There's a helper provided so that it doesn't need to be implemented per-class.  I should probably abstract this away entirely.
```suggestion
              "type" => self.class.type,
```